lib/open_api_spex/operation.ex

```diff
@@ -52,11 +52,39 @@ defmodule OpenApiSpex.Operation do
         }
 
   @doc """
-  Constructs an Operation struct from the plug and opts specified in the given route
+  Constructs an Operation struct from the plug and opts specified in the given route.
+
+  If any path parameters declared in the `route.path` do not have corresponding
+  parameters defined in the `Operation`, the result is an error tuple with a message
+  descring which parameters are missing.
   """
-  @spec from_route(PathItem.route) :: t
+  @spec from_route(PathItem.route) :: {:ok, t} | {:error, String.t()}
   def from_route(route) do
-    from_plug(route.plug, route.opts)
+    operation = from_plug(route.plug, route.opts)
+    check_all_path_params_declared(operation, route.path)
+  end
+
+  defp check_all_path_params_declared(operation, route_path) do
+    pattern = ~r/{(.*?)}/
```

Comment:
This is interesting. Why does open_api_spex have to parse routes itself? It could rely on Plug to do it, just like Phoenix does:

```elixir
iex(1)> Plug.Router.Utils.build_path_match("/foo/:id")
{[:id], ["foo", {:id, [], nil}]}
iex(2)> Plug.Router.Utils.build_path_match("/foo/bar/:id")
{[:id], ["foo", "bar", {:id, [], nil}]}
iex(3)> Plug.Router.Utils.build_path_match("/foo/:id/bar")
{[:id], ["foo", {:id, [], nil}, "bar"]}
```

I think that may be more robust.